extension/json/src/functions/table_functions/json_scan.cpp

```diff
@@ -245,12 +312,26 @@ static uint8_t* nextJson(uint8_t* ptr, uint64_t size) {
     return ptr == end ? nullptr : ptr;
 }
 
-void JSONScanLocalState::parseJson(uint8_t* jsonStart, uint64_t size, uint64_t remaining) {
+processor::WarningSourceData JSONScanLocalState::getWarningData(uint64_t startByteOffset,
+    uint64_t endByteOffset) const {
+    KU_ASSERT(currentBufferHandle);
+    return processor::WarningSourceData::constructFrom(currentBufferHandle->bufferIdx,
+        lineOrObjectCountInBuffer, startByteOffset, endByteOffset);
+}
+
+void JSONScanLocalState::handleParseError(yyjson_read_err& err, bool completedParsingObject,
+    const std::string& extra) const {
+    currentReader->throwParseError(err, completedParsingObject,
+        getWarningData(getFileOffset(), getFileOffset() + err.pos + 1), errorHandler.get(), extra);
+}
+
+void JSONScanLocalState::parseJson(uint8_t* jsonStart, uint64_t size, uint64_t remaining,
+    const std::optional<std::vector<ValueVector*>>& warningDataVectors) {
     yyjson_doc* doc = nullptr;
     yyjson_read_err err;
     doc = JSONCommon::readDocumentUnsafe(jsonStart, remaining, JSONCommon::READ_INSITU_FLAG, &err);
```

Comment:
I'm pretty sure this leaks memory because we never free the `doc`. I'll fix this in a separate PR.

Reply:
Ended up fixing it here anyways

Reply:
Opened PR [here](https://github.com/kuzudb/kuzu/pull/4302)

Reply:
That PR ended up not fixing everything 😭 so I ended fixing the rest here.

Reply:
Can u open an issue on running extension tests with asan?

Reply:
opened https://github.com/kuzudb/kuzu/issues/4312